### DISCUSSION TITLE
When Apollo Server v3 is out of life, is npm install impossible? Then, does nestJS using v3 break down as well?

### DISCUSSION THREAD

**Discussion by bedcoding** [apollographql/apollo-server#7198] [Q&A] [answered]
1. I saw the phrase that apollo server v3 is shutting down.
If Apollo Server v3 is deprecated, will all the libraries and frameworks that use it be unavailable for 'npm install'?
https://www.apollographql.com/docs/apollo-server/v3
![스크린샷 2022-11-28 오전 9 08 30](https://user-images.githubusercontent.com/33927558/204167299-883e5068-54ba-46b5-bcc9-76245d9b4725.jpg)

2. By the way, nestJS, which I am currently using, was also using v3 internally.
https://github.com/nestjs/nest/blob/master/package.json
![스크린샷 2022-11-28 오전 9 15 37](https://user-images.githubusercontent.com/33927558/204167718-19b8d231-0d6f-4e41-9d85-385307a08bed.jpg)

3. The official document of nestJS also instructs me to npm install the apollo-server-express that exists in the v3 version.
What should I do? I'm so afraid I don't know what I can do to prevent a disaster.
https://docs.nestjs.com/graphql/quick-start
![스크린샷 2022-11-28 오전 9 09 21](https://user-images.githubusercontent.com/33927558/204167861-e62233df-72c4-43ae-87aa-f3e2fad02a99.jpg)

4. Other official documents are still based on the v3 version.
Is there a case of updating nestJS and graphQL to the Apollo v4 version? (How do I update to Apollo v4 version?)
https://graphql.org/code/#javascript
![스크린샷 2022-11-28 오전 9 55 37](https://user-images.githubusercontent.com/33927558/204170637-2b0eda95-5dd9-4683-a3bf-1941277c3a76.jpg)

The apollo documentation says to use the '@apollo/server package'.
However, no matter how much I googled, I couldn't find any examples of using the @apollo/server package in a nestJS + graphQL project.
https://www.apollographql.com/docs/apollo-server/migration
![스크린샷 2022-11-28 오전 10 05 31](https://user-images.githubusercontent.com/33927558/204171100-ef66f65c-2ca7-4a82-8082-13ba0724f6d6.jpg)

**Comment by glasser** [comment, marked as answer]
We will not be deleting end-of-life versions from the npm repository, but you will also not be able to rely on security and other major fixes being applied, so upgrading in the next year would be an excellent idea.

As you've found, it looks like NestJS has an open issue for AS4 support at https://github.com/nestjs/graphql/issues/2445

My impression is the Nest team has been great about staying up to date with dependencies like AS in the past. If for some reason they are unable to upgrade because of an issue with AS4 I'm sure they will reach out to us and we can help!
